cpp/src/slice2py/PythonUtil.cpp

```diff
@@ -125,6 +125,24 @@ namespace
 
         return "???";
     }
+
+    /// Returns a DocString formatted link to the provided Slice identifier.
+    string pyLinkFormatter(string identifier, string memberComponent)
+    {
+        ostringstream os;
+        os << "`";
+        if !(identifier.empty())
+        {
+            os << Slice::Python::fixIdent(identifier);
+            if !(memberComponent.empty())
+            {
+                os << ".";
+            }
+        }
+        os << Slice::Python::fixIdent(memberComponent);
+        os << "`";
+        return os.str();
```

Comment:
We need to improve this, but I don't think the current callback provides enough context.

See the expected format https://www.sphinx-doc.org/en/master/usage/domains/python.html#cross-referencing-python-objects

Here to emit `:py:meth:`, `:py:class:`, ... we need to lookup the identifier. But lookup depends on the context where the links appear, and the link formatter doesn't have this context.

Reply:
Yes, there's a couple other languages with this problem.
And now that we have metadata to remap identifiers, lookup is essential in all languages.
Feel free to comment on the issue: https://github.com/zeroc-ice/ice/issues/3120